docs/DeveloperGuide.md

```diff
@@ -251,11 +251,18 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
-### \[Proposed\] Data archiving
+### Data archiving
 
-_{Explain here how the data archiving feature will be implemented}_
+The data archiving feature allows users to mark contacts as archived rather than permanently deleting them.
+Archiving can be used to manage inactive or unneeded contacts without losing historical information or
+requiring deletion, which may be irreversible and result in data loss.
```

Comment:
I think for this part can be phrased `irreversible after the app has been exited or closed.` for more clarity as it is reversible if the app is not exited yet. Also can consider adding a citing to user story for `archive` which says it is for ensuring contact list is focused on active cases.